CHANGELOG.md

```diff
@@ -2,6 +2,20 @@
 
 ## master / unreleased
 
+* [CHANGE] Updated the Ruler storage config to utilize Thanos `objstore.Bucket` instead of `chunk.ObjectClient` based clients. #2489
+  * Removed `-ruler.storage.azure.download-buffer-size`
+  * Removed `-ruler.storage.azure.upload-buffer-size`
+  * Added `-ruler.storage.azure.endpoint-suffix`
+  * Added `-ruler.storage.azure.max-retries`
+  * Changed `-ruler.storage.gcs.bucketname` > `-ruler.storage.gcs.bucket-name`
+  * Removed `-ruler.storage.gcs.chunk-buffer-size`
+  * Removed `-ruler.storage.gcs.request-timeout`
+  * Added `-ruler.storage.gcs.service-account`
+  * Changed `-ruler.storage.s3.url` > `-ruler.storage.s3.endpoint`, credentials are no longer url encoded and in-memory is no longer an option
+  * Changed `-ruler.storage.s3.buckets` > `-ruler.storage.s3.bucket-name`, multiple buckets can no longer be specified
+  * Added `-ruler.storage.s3.secret-access-key`
+  * Added `-ruler.storage.s3.access-key-id`
+  * Added `-ruler.storage.s3.insecure`
```

Comment:
Is this compatible with [1.x guarantees](https://cortexmetrics.io/docs/configuration/v1guarantees/)? It says that Ruler API is experimental, but Ruler itself is not.

> so we will will keep deprecated flags around for 2 minor release

Reply:
In order to use object storage with the ruler you need to use the experimental API. Since these configs are directly tied to the experimental API I think they should be safe to change.